core/src/main/scala/org/http4s/blaze/channel/nio1/NIO1SocketServerGroup.scala

```diff
@@ -0,0 +1,238 @@
+
+  override def closeGroup() {
+    logger.info("Closing NIO1SocketServerGroup")
+    isClosed = true
+    s.wakeup()
+  }
+
+
+  /** Create a [[ServerChannel]] that will serve the services on the requisite sockets */
+  override def bind(address: SocketAddress, service: BufferPipelineBuilder): Try[ServerChannel] = {
+    Try{
+      val ch = ServerSocketChannel.open().bind(address)
+      val serverChannel = new NIO1ServerChannel(ch, service)
+      t.listenOnChannel(serverChannel)
+
+      logger.info("Service bound to address " + ch.getLocalAddress)
+      serverChannel
+    }
+  }
+
+
+  private class AcceptThread extends Thread("NIO1SocketServerChannelFactory") {
+    setDaemon(true)
+
+  private val queue = new AtomicReference[List[NIO1ServerChannel]](Nil)
+
+    /** Add a channel to the selector loop */
+    def listenOnChannel(channel: NIO1ServerChannel) {
+      def go(): Unit = queue.get() match {
+        case null                                    => channel.close() // Queue is closed.
+        case q if queue.compareAndSet(q, channel::q) => s.wakeup()      // Successful set. Wake the loop.
+        case _                                       => go()            // Lost race. Try again.
+      }
+
+      go()
+    }
+
+    override def run(): Unit = {
+      while (!isClosed) {
+        s.select()      // wait for connections to come in
+
+        // Add any new connections
+        val q = queue.getAndSet(Nil)
+
+        q.foreach { ch =>
+          try {
+            ch.channel.configureBlocking(false)
+            ch.channel.register(s, SelectionKey.OP_ACCEPT, ch)
+
+          } catch {
+            case NonFatal(t) =>
+              logger.error(t)("Error during channel registration: " + ch.channel.getLocalAddress())
+              try ch.close()
+              catch { case NonFatal(_) => /* NOOP */ }
+          }
+        }
+
+        val it = s.selectedKeys().iterator()
+
+        while (it.hasNext()) {
+          val key = it.next()
+          it.remove()
+
+          val channel = key.attachment().asInstanceOf[NIO1ServerChannel]
+          val serverChannel = channel.channel
+          val service = channel.service
+          val loop = pool.nextLoop()
+
+          try {
+            val clientChannel = serverChannel.accept()
+
+            if (clientChannel != null) {                       // This should never be `null`
+            val address = clientChannel.getRemoteAddress()
+
+              // check to see if we want to keep this connection
+              if (acceptConnection(address)) {
+                clientChannel.setOption(java.net.StandardSocketOptions.TCP_NODELAY, java.lang.Boolean.FALSE)
+                loop.initChannel(service, clientChannel, key => new SocketChannelHead(clientChannel, loop, key))
+              }
+              else clientChannel.close()
+            }
+          }
+          catch {
+            case s: SecurityException => logger.info(s)("Connection rejected by SecurityManager.")
```

Comment:
Looking through source of Jetty, Netty, and Tomcat, I don't find any special treatment of this exception in the context of accepting connections. Therefor I believe you're right: this should be treated as an error.
Good catch.